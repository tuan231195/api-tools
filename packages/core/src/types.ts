import { OpenAPI, OpenAPIV2, OpenAPIV3 } from 'openapi-types';
import { Schema as MixSchema } from 'dtsgenerator';
import { HTTP_METHODS } from 'src/constants';
import type { JsonSchemaDraft07 } from 'dtsgenerator/src/core/jsonSchemaDraft07';

export type SchemaObject = OpenAPIV2.SchemaObject | OpenAPIV3.SchemaObject;

export type Parameter = OpenAPIV3.ParameterObject | OpenAPIV2.Parameter;

export type JsonSchema = JsonSchemaDraft07.SchemaObject;

export type Schema = Omit<MixSchema, 'content'> & {
	content: JsonSchema;
};

export const supportedMediaTypes = [
	'application/json',
	'application/x-www-form-urlencoded',
	'multipart/form-data',
] as const;

type SUPPORT_MEDIA_TYPE = typeof supportedMediaTypes[number];

export type ParsedOperationSchema = {
	params: Schema | null;
	queries: Schema | null;
	headers: Schema | null;
	requestBody: ParsedRequestBodySchema | null;
	responseBody: ParsedResponseBodySchema;
};

export type ParsedRequestBodySchema = {
	type: 'json' | 'form';
	contentType: SUPPORT_MEDIA_TYPE;
	schema: Schema;
};

export type ParsedResponseBodySchema = {
	statuses: Record<number, Schema>;
	all: Schema | null;
	success: Schema | null;
};

export type HttpMethod = typeof HTTP_METHODS[number];

export type Operation = {
	method: HttpMethod;
	path: string;
	operationId: string;
	operation: OpenAPI.Operation;
};

export type ParsedOperationSchemaWithInfo = {
	info: Operation;
	schema: ParsedOperationSchema;
};

export type GetOperationArgs =
	| {
			operationId: string;
	  }
	| {
			methodName: HttpMethod;
			path: string;
	  };

export type ParsedSchema = Record<string, ParsedOperationSchemaWithInfo>;

export type ParsedDocument = {
	document: OpenAPI.Document;
	schema: ParsedSchema;
};
