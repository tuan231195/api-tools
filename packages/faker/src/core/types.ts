import { MergeWithCustomizer } from 'lodash';
import { HttpMethod } from '@vdtn359/api-tools-core';

export type FakeOperation = {
	headers: any;
	params: any;
	queries: any;
	responseBody: any;
	requestBody: any;
};

export type OverrideOptions = {
	overrides?: any | ((fakeData: any) => any);
	mergeOptions?: MergeWithCustomizer;
};

export type OperationOverrideOptions = OverrideOptions &
	(
		| {
				operationId: string;
		  }
		| {
				methodName: HttpMethod;
				path: string;
		  }
	);
