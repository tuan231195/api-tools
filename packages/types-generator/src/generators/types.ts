import { parse } from '@vdtn359/api-tools-core';
import path from 'path';
import { kebabCase } from 'lodash';
import logger from 'src/logger';
import fs from 'fs-extra';
import { prepare, prettify, toTsTypes } from 'src/generators/utils';

export const generateTypes = async (uri: string, outDir: string) => {
	const parsedSchema = await parse(uri);
	await prepare(outDir);
	await Promise.all(
		Object.entries(parsedSchema).map(
			async ([operationId, { schema: operationSchema }]) => {
				const writeFile = path.resolve(
					outDir,
					`${kebabCase(operationId)}.ts`
				);
				logger(`Write ${operationId} to file ${writeFile}`);

				const content = await toTsTypes(operationSchema);
				await fs.writeFile(writeFile, prettify(content));
			}
		)
	);
};
