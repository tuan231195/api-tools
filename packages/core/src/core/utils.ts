import { zipObject } from 'lodash';

type Awaited<T> = T extends PromiseLike<infer U> ? Awaited<U> : T;

type ResolvedPromise<T extends Record<string, any>> = {
	[k in keyof T]: Awaited<T[k]>;
};

export const promiseAll = async <T extends Record<string, any>>(
	object: T
): Promise<ResolvedPromise<T>> => {
	const keys = Object.keys(object);
	const resolvedArray = await Promise.all(keys.map((key) => object[key]));
	return zipObject(keys, resolvedArray) as ResolvedPromise<T>;
};
