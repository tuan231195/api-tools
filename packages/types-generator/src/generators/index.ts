export * from './axios';
export * from './express';
export * from './types';
