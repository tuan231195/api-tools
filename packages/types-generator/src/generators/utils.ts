import prettier from 'prettier';
import fs from 'fs-extra';
import {
	generate,
	ParsedSchemaOperation,
	Schema,
} from '@vdtn359/api-tools-core';
import os from 'os';

export const prettify = (content: string) => {
	return prettier.format(content, {
		semi: true,
		parser: 'typescript',
		useTabs: true,
		tabWidth: 4,
		singleQuote: true,
	});
};

export const prepare = async (outDir: string) => {
	await fs.mkdirp(outDir);
};

export const toTsType = async (schema: Schema | null) => {
	if (!schema) {
		return null;
	}
	return toTsTypes([schema]);
};

export const toTsTypes = async (schema: ParsedSchemaOperation | Schema[]) => {
	return (await generate(schema))
		.replace(/^declare interface/gm, 'export interface')
		.replace(/^declare type/gm, 'export type')
		.replace(/(\r?\nexport)/g, (match) => `${os.EOL}${match}`);
};
