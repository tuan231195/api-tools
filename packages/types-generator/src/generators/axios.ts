import {
	parse,
	ParsedSchema,
	ParsedSchemaOperation,
	promiseAll,
	Schema,
	validate,
} from '@vdtn359/api-tools-core';
import fs from 'fs-extra';
import path from 'path';
import logger from 'src/logger';
import { kebabCase } from 'lodash';
import { compileTemplate } from 'src/templates';
import { prepare, prettify, toTsType, toTsTypes } from 'src/generators/utils';

const operationTemplate = compileTemplate(
	path.resolve(__dirname, '..', 'templates', 'axios', 'operation.hbs')
);

const indexTemplate = compileTemplate(
	path.resolve(__dirname, '..', 'templates', 'axios', 'index.hbs')
);

export const generateAxiosTypes = async (uri: string, outDir: string) => {
	const document = await validate(uri);
	const parsedSchema = await parse(document);
	await prepare(outDir);

	await generateOperations(parsedSchema, outDir);
	await generateIndex(parsedSchema, outDir);
};

const generateIndex = async (parsedSchema: ParsedSchema, outDir: string) => {
	const operations = Object.entries(parsedSchema).map(
		([operationId, { schema: operationSchema }]) => {
			const successResponses = getSuccessResponseSchema(operationSchema);
			return {
				hasResponse: successResponses.length > 0,
				operationId,
			};
		}
	);
	const writeFile = path.resolve(outDir, 'index.ts');
	logger(`Write index to file ${writeFile}`);

	const templateContent = indexTemplate({
		operations,
	});
	await fs.writeFile(writeFile, prettify(templateContent));
};

const generateOperations = async (
	parsedSchema: ParsedSchema,
	outDir: string
) => {
	await Promise.all(
		Object.entries(parsedSchema).map(
			async ([operationId, { schema: operationSchema, operation }]) => {
				try {
					const writeFile = path.resolve(
						outDir,
						`${kebabCase(operationId)}.ts`
					);
					logger(`Write ${operationId} to file ${writeFile}`);

					const successResponses =
						getSuccessResponseSchema(operationSchema);

					const templateContent = operationTemplate(
						promiseAll({
							url: createUrl(operation.path),
							httpMethod: operation.method,
							hasHeaders: !!operationSchema.headers,
							hasParams: !!operationSchema.params,
							hasQueries: !!operationSchema.queries,
							hasRequestBody: !!operationSchema.requestBody,
							hasResponse: successResponses.length > 0,
							requestBodyType: operationSchema.requestBody
								? toTsType(operationSchema.requestBody.schema)
								: '',
							isForm:
								operationSchema.requestBody?.type === 'form',
							responseType: successResponses.length
								? toTsTypes(successResponses)
								: '',
							paramsType: toTsType(operationSchema.params),
							queriesType: toTsType(operationSchema.queries),
							headersType: toTsType(operationSchema.headers),
						})
					);
					await fs.writeFile(writeFile, prettify(templateContent));
				} catch (e) {
					console.error(
						`Failed to generate types for ${operationId}`,
						e
					);
				}
			}
		)
	);
};

const getSuccessResponseSchema = (schema: ParsedSchemaOperation) => {
	const successResponses: Schema[] = Object.entries(
		schema.responseBody.statuses
	)
		.filter(([status]) => {
			const statusCode = parseInt(status, 10);
			return statusCode >= 200 && statusCode < 300;
		})
		.map(([, schema]) => schema);
	if (schema.responseBody.success) {
		schema.responseBody.success.content.$id = 'Response';
		successResponses.push(schema.responseBody.success);
	}
	return successResponses;
};

export const createUrl = (pathName: string): string => {
	return pathName
		.split('/')
		.map((item) => {
			if (item.startsWith('{')) {
				return `\${encodeURIComponent(request.path['${item.substring(
					1,
					item.length - 1
				)}'] as any)}`;
			}
			return item;
		})
		.join('/');
};
