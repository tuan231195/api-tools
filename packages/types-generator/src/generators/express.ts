import {
	OpenAPI,
	parse,
	ParsedSchema,
	Schema,
	validate,
} from '@vdtn359/api-tools-core';
import fs from 'fs-extra';
import path from 'path';
import logger from 'src/logger';
import { kebabCase } from 'lodash';
import { compileTemplate } from 'src/templates';
import { prepare, prettify, toTsType, toTsTypes } from 'src/generators/utils';

const handlerTemplate = compileTemplate(
	path.resolve(__dirname, '..', 'templates', 'express', 'handler.hbs')
);

export const generateExpressTypes = async (uri: string, outDir: string) => {
	const document = await validate(uri);
	const parsedSchema = await parse(document);
	await prepare(outDir);

	await generateHandlers(document, parsedSchema, outDir);
};

const generateHandlers = async (
	document: OpenAPI.Document,
	parsedSchema: ParsedSchema,
	outDir: string
) => {
	await Promise.all(
		Object.entries(parsedSchema).map(
			async ([operationId, { schema: operationSchema }]) => {
				try {
					const writeFile = path.resolve(
						outDir,
						`${kebabCase(operationId)}.ts`
					);
					logger(`Write ${operationId} to file ${writeFile}`);

					const allStatuses = Object.entries(
						operationSchema.responseBody.statuses
					);
					const templateContent = handlerTemplate({
						hasHeaders: !!operationSchema.headers,
						hasParams: !!operationSchema.params,
						hasQueries: !!operationSchema.queries,
						hasRequestBody: !!operationSchema.requestBody,
						hasResponse: !!operationSchema.responseBody.all,
						requestBodyType: await toTsType(
							operationSchema.requestBody
						),
						allResponses: await toTsTypes(
							[
								...allStatuses.map(([, schema]) => schema),
								operationSchema.responseBody.all,
							].filter(Boolean) as Schema[]
						),
						responseTypes: allStatuses.map(([status, schema]) => ({
							statusCode: status,
							name: schema.content.$id,
						})),
						paramsType: await toTsType(operationSchema.params),
						queriesType: await toTsType(operationSchema.queries),
						headersType: await toTsType(operationSchema.headers),
					});
					await fs.writeFile(writeFile, prettify(templateContent));
				} catch (e) {
					console.error(
						`Failed to generate types for ${operationId}`,
						e
					);
				}
			}
		)
	);
};
