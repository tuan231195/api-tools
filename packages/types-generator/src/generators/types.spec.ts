import path from 'path';
import { matchFsSnapshot } from '#test/utils';
import { generateTypes } from 'src/generators/types';

const testDir = path.resolve(__dirname, '..', '..', '..', '..', 'test');

jest.mock('fs-extra', () => {
	const { fs } = require('memfs');
	const util = require('util');
	const originalFs = jest.requireActual('fs-extra');

	return {
		...originalFs,
		mkdirp: util.promisify(fs.mkdirp.bind(fs)),
		writeFile: util.promisify(fs.writeFile.bind(fs)),
	};
});

describe('generateTypescriptTypes', () => {
	it.each([
		['/v2/petstore.json', path.resolve(testDir, 'v2', 'petstore.json')],
		['/v3/petstore.json', path.resolve(testDir, 'v3', 'petstore.json')],
	])('%s', async (testName, filePath) => {
		await generateTypes(filePath, 'out');
		await matchFsSnapshot('out');
	});
});
