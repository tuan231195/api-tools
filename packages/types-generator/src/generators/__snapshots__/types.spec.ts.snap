// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateTypescriptTypes /v2/petstore.json 1`] = `
Object {
  "add-pet.ts": "export interface RequestBody {
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}

export type Response405 = void;
",
  "create-user.ts": "export interface RequestBody {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}

export type Response200 = void;
",
  "create-users-with-array-input.ts": "export type RequestBody = {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}[];

export type Response200 = void;
",
  "create-users-with-list-input.ts": "export type RequestBody = {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}[];

export type Response200 = void;
",
  "delete-order.ts": "export interface Path {
	/**
	 * ID of the order that needs to be deleted
	 */
	orderId: string; // uuid
}

export type Response400 = void;

export type Response404 = void;
",
  "delete-pet.ts": "export interface Header {
	api_key?: string;
}

export interface Path {
	/**
	 * Pet id to delete
	 */
	petId: string; // uuid
}

export type Response400 = void;

export type Response404 = void;
",
  "delete-user.ts": "export interface Path {
	/**
	 * The name that needs to be deleted
	 */
	username: string;
}

export type Response400 = void;

export type Response404 = void;
",
  "find-pets-by-status.ts": "export interface Query {
	/**
	 * Status values that need to be considered for filter
	 */
	status: ('available' | 'pending' | 'sold')[];
}

export type Response200 = {
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}[];

export type Response400 = void;
",
  "find-pets-by-tags.ts": "export interface Query {
	/**
	 * Tags to filter by
	 */
	tags: string[];
}

export type Response200 = {
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}[];

export type Response400 = void;
",
  "get-inventory.ts": "export interface Response200 {
	[name: string]: number; // int32
}
",
  "get-order-by-id.ts": "export interface Path {
	/**
	 * ID of pet that needs to be fetched
	 */
	orderId: string; // uuid
}

export interface Response200 {
	id?: string; // uuid
	petId?: string; // uuid
	quantity?: number; // int32
	shipDate?: string; // date-time
	/**
	 * Order Status
	 * example:
	 * approved
	 */
	status?: 'placed' | 'approved' | 'delivered';
	complete?: boolean;
}

export type Response400 = void;

export type Response404 = void;
",
  "get-pet-by-id.ts": "export interface Path {
	/**
	 * ID of pet to return
	 */
	petId: string; // uuid
}

export interface Response200 {
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}

export type Response400 = void;

export type Response404 = void;
",
  "get-user-by-name.ts": "export interface Path {
	/**
	 * The name that needs to be fetched. Use user1 for testing.
	 */
	username: string;
}

export interface Response200 {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}

export type Response400 = void;

export type Response404 = void;
",
  "login-user.ts": "export interface Query {
	/**
	 * The user name for login
	 */
	username: string;
	/**
	 * The password for login in clear text
	 */
	password: string;
}

export type Response200 = string;

export type Response400 = void;
",
  "logout-user.ts": "export type Response200 = void;
",
  "place-order.ts": "export interface RequestBody {
	id?: string; // uuid
	petId?: string; // uuid
	quantity?: number; // int32
	shipDate?: string; // date-time
	/**
	 * Order Status
	 * example:
	 * approved
	 */
	status?: 'placed' | 'approved' | 'delivered';
	complete?: boolean;
}

export interface Response200 {
	id?: string; // uuid
	petId?: string; // uuid
	quantity?: number; // int32
	shipDate?: string; // date-time
	/**
	 * Order Status
	 * example:
	 * approved
	 */
	status?: 'placed' | 'approved' | 'delivered';
	complete?: boolean;
}

export type Response400 = void;
",
  "update-pet-with-form.ts": "export interface Path {
	/**
	 * ID of pet that needs to be updated
	 */
	petId: string; // uuid
}

export interface RequestBody {
	/**
	 * Updated name of the pet
	 */
	name?: string;
	/**
	 * Updated status of the pet
	 */
	status?: string;
}

export type Response405 = void;
",
  "update-pet.ts": "export interface RequestBody {
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}

export type Response400 = void;

export type Response404 = void;

export type Response405 = void;
",
  "update-user.ts": "export interface Path {
	/**
	 * name that need to be updated
	 */
	username: string;
}

export interface RequestBody {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}

export type Response400 = void;

export type Response404 = void;
",
  "upload-file.ts": "export interface Path {
	/**
	 * ID of pet to update
	 */
	petId: string; // uuid
}

export interface RequestBody {
	/**
	 * Additional data to pass to server
	 */
	additionalMetadata?: string;
	/**
	 * file to upload
	 */
	file?: unknown;
}

export interface Response200 {
	code?: number; // int32
	type?: string;
	message?: string;
}
",
}
`;

exports[`generateTypescriptTypes /v3/petstore.json 1`] = `
Object {
  "add-pet.ts": "export interface RequestBody {
	/**
	 * example:
	 * 10
	 */
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}

export interface Response200 {
	/**
	 * example:
	 * 10
	 */
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}

export type Response405 = void;
",
  "create-user.ts": "export interface RequestBody {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}

export interface Response200 {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}
",
  "create-users-with-array-input.ts": "export type RequestBody = {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}[];

export type Response200 = void;
",
  "create-users-with-list-input.ts": "export type RequestBody = {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}[];

export type Response200 = void;
",
  "delete-order.ts": "export interface Path {
	orderId: string; // uuid
}

export type Response400 = void;

export type Response404 = void;
",
  "delete-pet.ts": "export interface Header {
	api_key?: string;
}

export interface Path {
	petId: string; // uuid
}

export type Response400 = void;
",
  "delete-user.ts": "export interface Path {
	username: string;
}

export type Response400 = void;

export type Response404 = void;
",
  "find-pets-by-status.ts": "export interface Query {
	status?: 'available' | 'pending' | 'sold';
}

export type Response200 = {
	/**
	 * example:
	 * 10
	 */
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}[];

export type Response400 = void;
",
  "find-pets-by-tags.ts": "export interface Query {
	tags?: string[];
}

export type Response200 = {
	/**
	 * example:
	 * 10
	 */
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}[];

export type Response400 = void;
",
  "get-inventory.ts": "export interface Response200 {
	[name: string]: number; // int32
}
",
  "get-order-by-id.ts": "export interface Path {
	orderId: string; // uuid
}

export interface Response200 {
	id?: string; // uuid
	petId?: string; // uuid
	quantity?: number; // int32
	shipDate?: string; // date-time
	/**
	 * Order Status
	 * example:
	 * approved
	 */
	status?: 'placed' | 'approved' | 'delivered';
	complete?: boolean;
}

export type Response400 = void;

export type Response404 = void;
",
  "get-pet-by-id.ts": "export interface Path {
	petId: string; // uuid
}

export interface Response200 {
	/**
	 * example:
	 * 10
	 */
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}

export type Response400 = void;

export type Response404 = void;
",
  "get-user-by-name.ts": "export interface Path {
	username: string;
}

export interface Response200 {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}

export type Response400 = void;

export type Response404 = void;
",
  "login-user.ts": "export interface Query {
	username?: string;
	password?: string;
}

export type Response200 = string;

export type Response400 = void;
",
  "logout-user.ts": "export type Response200 = void;
",
  "place-order.ts": "export interface RequestBody {
	id?: string; // uuid
	petId?: string; // uuid
	quantity?: number; // int32
	shipDate?: string; // date-time
	/**
	 * Order Status
	 * example:
	 * approved
	 */
	status?: 'placed' | 'approved' | 'delivered';
	complete?: boolean;
}

export interface Response200 {
	id?: string; // uuid
	petId?: string; // uuid
	quantity?: number; // int32
	shipDate?: string; // date-time
	/**
	 * Order Status
	 * example:
	 * approved
	 */
	status?: 'placed' | 'approved' | 'delivered';
	complete?: boolean;
}

export type Response405 = void;
",
  "update-pet-with-form.ts": "export interface Path {
	petId: string; // uuid
}

export interface Query {
	name?: string;
	status?: string;
}

export type Response405 = void;
",
  "update-pet.ts": "export interface RequestBody {
	/**
	 * example:
	 * 10
	 */
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}

export interface Response200 {
	/**
	 * example:
	 * 10
	 */
	id?: string; // uuid
	/**
	 * example:
	 * doggie
	 */
	name: string;
	category?: {
		id?: string; // uuid
		name?: string;
	};
	photoUrls: string /* uri */[];
	tags?: {
		id?: string; // uuid
		name?: string;
	}[];
	/**
	 * pet status in the store
	 */
	status?: 'available' | 'pending' | 'sold';
}

export type Response400 = void;

export type Response404 = void;

export type Response405 = void;
",
  "update-user.ts": "export interface Path {
	username: string;
}

export interface RequestBody {
	id: string; // uuid
	/**
	 * example:
	 * theUser
	 */
	username: string;
	/**
	 * example:
	 * John
	 */
	firstName: string;
	/**
	 * example:
	 * James
	 */
	lastName: string;
	/**
	 * example:
	 * john@email.com
	 */
	email?: string;
	/**
	 * example:
	 * 12345
	 */
	password: string;
	/**
	 * example:
	 * 12345
	 */
	phone?: string;
	/**
	 * User Status
	 */
	userStatus?: boolean;
}

export interface Response200 {
	id: string; // uuid
}
",
  "upload-file.ts": "export interface Path {
	petId: string; // uuid
}

export interface Query {
	additionalMetadata?: string;
}

export interface Response200 {
	code?: number; // int32
	type?: string;
	message?: string;
}
",
}
`;
